packages/wallet/src/redux/channel-store/reducer.ts

```diff
@@ -2,21 +2,38 @@ import { ChannelStore, getChannel, setChannel } from './state';
 
 import { ReducerWithSideEffects } from '../../utils/reducer-utils';
 import { StateWithSideEffects } from '../utils';
-import { WalletAction } from '../actions';
 import {
   SignedCommitment,
   Commitment,
   signCommitment2,
   getChannelId,
   hasValidSignature,
 } from '../../domain';
-import { pushCommitment, ChannelState } from './channel-state/states';
+import { pushCommitment, ChannelState, initializeChannel } from './channel-state/states';
 import { validTransition } from './channel-state';
+import * as channelActions from './actions';
 
-export const channelStateReducer: ReducerWithSideEffects<ChannelStore> = (
+export const channelStoreReducer: ReducerWithSideEffects<ChannelStore> = (
   state: ChannelStore,
-  action: WalletAction,
+  action: channelActions.ChannelAction,
 ): StateWithSideEffects<ChannelStore> => {
+  switch (action.type) {
+    case channelActions.OPPONENT_COMMITMENT_RECEIVED:
```

Comment:
This was added to support existing protocols that rely on the channel state reducer. If we switch everything to `signAndStore`/`checkAndStore` these can go,